enhancements/single-node-production-edge-cluster-profile.md

```diff
@@ -0,0 +1,380 @@
+  this cluster profile.
+
+One example of this use case is seen in telecommunication service
+providers implementation of a Radio Access Network (RAN). This use case
+is discussed in more detail below.
+
+## Motivation
+
+The benefits of the cloud native approach to developing and deploying
+applications is increasingly being adopted in the context of edge
+computing. Here we see that as the the distance between an site and the
+central management hub grows, the number of servers at the site tends to
+shrink. The most distant sites typically have physical space for 1 server.
+
+We are seeing an emerging pattern in which some infrastructure providers and application
+owners desire:
+
+1. A consistent deployment approach for their workloads across these
+   disparate environments.
+3. That the edge sites can operate independantly from the central
+   management hub.
+
+And so, these users who have adopted Kubernetes at their their central
+management sites wish to have independent Kubernetes clusters at the
+more remote sites. 
+
+Of the several options explored for supporting the use of Kubernetes
+patterns for managing workloads at these sites (see the alternatives
+listed blow) a single-node deployment profile of OpenShift is the best
+way to give users a consistent experience across all of their sites.
+
+### Radio Access Network (RAN) Use Case
+
+In the context of telcocommunications service providers' 5G Radio Access
+Networks, it is increasingly common to see "cloud native" implementations
+of the 5G Distributed Unit (DU) component. Due to latency constraints,
+this DU component needs to be deployed very close to the radio antenna for
+which it is responsible. In practice, this can mean running this
+component on anything from a single server at the base of a remote cell
+tower or in a datacenter-like environment serving several base stations.
+
+A hypothetical DU example is an unusually resource-intensive workload,
+requiring 20 dedicated cores, 24 GiB of RAM consumed as huge pages,
+multiple SR-IOV NICs carrying several Gbps of traffic each, and
+specialized accelerator devices. The node hosting this workload must
+run a realtime
+kernel, be carefully tuned to ensure low-latency requirements
+can be met, and be configured to support features like Precision Timing
+Protocol (PTP).
+
+One crucial detail of this use case is the "cloud" hosting this workload
+is expected to be "autonomous" such that it can continue operating with
+its existing configuration and running the existing workload, even when
+any centralized management functionality is unavailable.
+
+### Goals
+
+* This enhancement describes an approach for deploying OpenShift in
+  single-node configurations for production use in environments with
+  "reasonably significant" memory, storage, and compute resources.
+* Clusters built using the `single-node-production-edge` profile
+  should pass most Kubernetes and OpenShift conformance end-to-end
```

Comment:
Yeah, that's the general idea. These sorts of deployments are expected to be part of a larger system that includes an orchestration tool outside of the cluster for managing 100s or 1000s of individual instances. Assuming a wipe-and-rebuild approach can be implemented within the other constraints like the length of change windows, it seemed reasonable to go that route instead of trying to make in-place work. On the other hand, if in-place isn't a big deal, maybe we don't need to make that assumption. So, it's definitely still up for discussion.

Reply:
We still need to do a lot of that analysis. The goal is to be as close as possible to the default deployment.

Reply:
There was some discussion about the topic in https://github.com/openshift/enhancements/pull/482/#discussion_r503354303 I'm guessing the goal for both of these will be identical, since in both cases you're running a limited set of functionality, iow. only core elements w/o addons.